ParallelVTK.jl

```diff
@@ -0,0 +1,45 @@
+using MPI
+using WaterLily
+using WriteVTK
+using Printf: @sprintf
+
+include("WaterLilyMPI.jl")
+
+components_first(a::AbstractArray{T,N}) where {T,N} = permutedims(a,[N,1:N-1...])
+
+"""Flow around a circle"""
+function circle(n,m,center,radius;Re=250,U=1,psolver=Poisson)
+    body = AutoBody((x,t)->√sum(abs2, x .- center) - radius)
+    Simulation((n,m), (U,0), radius; ν=U*radius/Re, body, psolver=psolver)
+end
+
+function write!(w,a::Simulation)
+    k = w.count[1]; N = size(inside(sim.flow.p))
+    xs = Tuple(ifelse(x==0,1,x+3):ifelse(x==0,n+4,n+x+6) for (n,x) in zip(N,grid_loc()))
+    extents = MPI.Allgather(xs, mpi_grid().comm)
```

Comment:
this can be done upon initializing the writer to avoid global operation at every write.